### PR TITLE
Better import for sometimes failing mypy test

### PR DESCRIPTION
**Pull request by mpharrigan**
Would get:

dev_tools/bash_scripts_test.py:33: error: Name
'_pytest.tmpdir.TempdirFactory' is not defined

This fixes it. Don't know why it sometimes works on
some configurations.

### CODE REVIEW

File: dev_tools/bash_scripts_test.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 from dev_tools import shell_tools
 
 if TYPE_CHECKING:
-    import _pytest
+    import _pytest.tmpdir
```
Comment (95-martin-orion) [inline dev_tools/bash_scripts_test.py:21]: Do you have a sense for _why_ this works? I don't really understand the root problem, even - my inuition is that `import _pytest` should give access to `_pytest.tmpdir.TempdirFactory` in the original code.

Comment (mpharrigan) [reply dev_tools/bash_scripts_test.py:21]: In python, `import package` does not give you availability to `package.subpackage`. But sometimes people like us put imports into the top-level `__init__.py`.

Comment (mpharrigan) [reply dev_tools/bash_scripts_test.py:21]: maybe there are different versions of pytest (`_pytest`?) around.

Comment (balopat) [reply dev_tools/bash_scripts_test.py:21]: The `_pytest` package does not have `tmpdir` attribute: https://github.com/pytest-dev/pytest/blob/main/src/_pytest/__init__.py

This should never work in mypy...but sometimes it does! :) 

```
>>> import _pytest
>>> _pytest.tmpdir
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module '_pytest' has no attribute 'tmpdir'
>>> import _pytest.tmpdir
>>> _pytest.tmpdir
<module '_pytest.tmpdir' from '/Users/balintp/.virtualenvs/cirq37/lib/python3.7/site-packages/_pytest/tmpdir.py'>

```

I can confirm that this fixes the mypy issue on my 3.7 mac env where I could consistently reproduce the error report.

**Review by balopat — APPROVED** [review]
Thanks for the fix!!!

File: dev_tools/bash_scripts_test.py (hunk 1 above)
Comment (mpharrigan) [reply dev_tools/bash_scripts_test.py:21]: `_pytest.*` is listed in mypy.ini as `follow_imports=silent` and `ignore_missing_imports=true`. I don't want to actually do any sleuthing but it sounds relevant and could have been a corner case whose behavior has changed between versions.

**Comment by mpharrigan** [comment]
This change got reverted somehow

**Comment by 95-martin-orion** [comment]
> This change got reverted somehow

Not only that, this file got reverted all the way to its version from December...

**Comment by balopat** [comment]
Hmmm...my "automated rebasing" strategy was error prone, this must have been overridden then. Let me fix that.

**Pull request merged**